server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java

```diff
@@ -400,8 +400,7 @@ private void validateServerFamily(ArrayList<Short> violations, ColumnUpdate colu
         String lockId = new String(columnUpdate.getValue(), UTF_8);
 
         try {
-          lockHeld = ServiceLock.isLockHeld(context.getZooCache(),
-              new ZooUtil.LockID(context.getZooKeeperRoot(), lockId));
+          lockHeld = ServiceLock.isLockHeld(context.getZooCache(), new ZooUtil.LockID("", lockId));
```

Comment:
Seems like LockId used to strip the root and add it back.  Now its mostly stripping `/` and its kinda confusing, would be good to clean this behavior up in a follow on PR.

Reply:
I completely agree. It's very confusing to me as well. I made an attempt to improve it, but I wasn't really sure the full intent of the code paths in the serialization/deserialization, as the intent of the current behavior is not documented.

Reply:
It may have been a crude form of compression, removing the root path from persisted data because it was always the same.

Reply:
I noticed that some representations also omit the leading slash, so it's not clear to me if it the `/root/` is always stripped, or only `/root`. I also don't know if all the places where this is persisted, to ensure we cover all the edge cases during an upgrade.

Reply:
Created #5410